ygen/protogen.go

```diff
@@ -941,12 +1104,19 @@ func genListKeyProto(listPackage string, listName string, args protoDefinitionAr
 		km.Fields = append(km.Fields, fd)
 		ctag++
 	}
-
-	p, _ := stripPackagePrefix(args.parentPackage, listPackage)
-	ltype := fmt.Sprintf("%s.%s", p, listName)
-	if listPackage == "" {
-		// Handle the case that the context of the list is already the base package.
+	var ltype string
+	if args.nestedMessages {
+		// Since the protobuf resolution rules mean that the parent scope is
+		// searched, then we do not need to qualify the name of the list message,
+		// even though it is in the parent's namespace.
 		ltype = listName
+	} else {
```

Comment:
Consider refactoring to make this the if case, with ltype := listName above the if.

Reply:
Done - yes. Not sure what I was thinking here.